activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -426,16 +426,14 @@ def pk_and_sequence_for(table) #:nodoc:
 
         def primary_keys(table_name) # :nodoc:
           select_values(<<-SQL.strip_heredoc, "SCHEMA")
-            WITH pk_constraint AS (
-              SELECT conrelid, unnest(conkey) AS connum FROM pg_constraint
-              WHERE contype = 'p'
-                AND conrelid = #{quote(quote_table_name(table_name))}::regclass
-            ), cons AS (
-              SELECT conrelid, connum, row_number() OVER() AS rownum FROM pk_constraint
-            )
-            SELECT attr.attname FROM pg_attribute attr
-            INNER JOIN cons ON attr.attrelid = cons.conrelid AND attr.attnum = cons.connum
-            ORDER BY cons.rownum
+                SELECT a.attname
+                  FROM pg_index i
+            CROSS JOIN generate_subscripts(i.indkey, 1) k
+                  JOIN pg_attribute a
+                    ON a.attrelid=i.indrelid
+                   AND a.attnum=i.indkey[k]
+                 WHERE i.indrelid=#{quote(quote_table_name(table_name))}::regclass
```

Comment:
[nit] I prefer space around equals even in SQL like `Style/SpaceAroundOperators`.
https://github.com/rails/rails/blob/55d66e214825b89442383f90adf575cf8c96568c/.rubocop.yml#L66-L67

We have few lines missing space around equals.

```
% git grep -n "\w=[\w#']" lib/active_record/connection_adapters
lib/active_record/connection_adapters/abstract_mysql_adapter.rb:556:        raw_table_options.sub!(/(ENGINE=\w+)(?: AUTO_INCREMENT=\d+)/, '\1')
lib/active_record/connection_adapters/abstract_mysql_adapter.rb:561:        if raw_table_options.sub!(/ COMMENT='.+'/, "")
lib/active_record/connection_adapters/abstract_mysql_adapter.rb:845:          # to work with MySQL 5.7.6 which sets optimizer_switch='derived_merge=on'
lib/active_record/connection_adapters/sqlite3_adapter.rb:348:            WHERE name=#{quote(row['name'])} AND type='index'
lib/active_record/connection_adapters/sqlite3_adapter.rb:352:            WHERE name=#{quote(row['name'])} AND type='index'

% git grep -n "}=[\w#']" lib/active_record/connection_adapters
lib/active_record/connection_adapters/abstract_mysql_adapter.rb:486:          execute("UPDATE #{quote_table_name(table_name)} SET #{quote_column_name(column_name)}=#{quote(default)} WHERE #{quote_column_name(column_name)} IS NULL")
lib/active_record/connection_adapters/postgresql/schema_statements.rb:528:            execute("UPDATE #{quote_table_name(table_name)} SET #{quote_column_name(column_name)}=#{quote_default_expression(default, column)} WHERE #{quote_column_name(column_name)} IS NULL") if column
lib/active_record/connection_adapters/sqlite3_adapter.rb:418:          exec_query("UPDATE #{quote_table_name(table_name)} SET #{quote_column_name(column_name)}=#{quote(default)} WHERE #{quote_column_name(column_name)} IS NULL")
```